src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setFilter: React.Dispatch<React.SetStateAction<Filter>>;
+  filter: Filter;
+  todos: Todo[];
+  setTodosToDelete: React.Dispatch<React.SetStateAction<number[]>>;
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  setFilter,
+  filter,
+  todos,
+  setTodosToDelete,
+}) => {
+  const todosLeft = todos.filter(todo => !todo.completed).length;
+
+  const handeClearCompleted = () => {
+    const completed = todos.filter(t => t.completed);
+    const ids: number[] = [];
+
+    completed.map(t => ids.push(t.id));
+    setTodosToDelete(ids);
+  };
+
+  return (
+    <>
+      {!!todos.length && (
```

Comment:
redundant condition, you already have one in App.tsx component, also you can remove react fragment.